### ISSUE TITLE
callProcess exception undocumented

### ISSUE THREAD

**Issue by rightfold** [haskell/process#60]
The docs say that callProcess throws an exception when the process returns non-zero. It is not documented what type of exception this is and therefore this API is pretty much useless. Please consider documenting the exception type or adding a new API that returns the status code instead.

**Comment by snoyberg** [comment]
A PR for a doc improvement would definitely be appreciated.

**Comment by rightfold** [comment]
Similar problems exist with most other functions exported by this module.

**Comment by rightfold** [comment]
@snoyberg I'll see what I can do :) but it won't be today, probably somewhere this weekend

**Comment by Profpatsch** [comment]
Same. “an Exception is thrown” is hardly very helpful documentation.

**Comment by Profpatsch** [comment]
I’d create a PR if I could find the origin of `OtherError`.

**Comment by snoyberg** [comment]
It's generated by `processFailedException`, which is used in a few functions in `System.Process`.

**Comment by Profpatsch** [comment]
Yes, that is pretty clear with a search, but there is no definition of `OtherError` anywhere, and Hoogle/Hayoo can’t find any as well.
